File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        nums_count = Counter(nums)
        heap = []

        for num,freq in nums_count.items():
            heapq.heappush(heap,(freq, num))
            while len(heap) > k:
                heapq.heappop(heap)
            
        return [num for freq,num in heap]

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        
        int l = 0;
        int r = nums.length-1;
        int mid = 0;
        
        while(l < r){
            mid = l + (r-l)/2;
            if(nums[mid] < nums[r])
                r = mid;
            else 
                l = mid + 1;
        }
        return nums[l];
    }
}
